presto-genericjdbc/src/main/java/io/prestosql/plugin/genericjdbc/GenericJdbcClientModule.java

```
@@ -0,0 +1,80 @@
+import io.prestosql.plugin.jdbc.ConnectionFactory;
+import io.prestosql.plugin.jdbc.DriverConnectionFactory;
+import io.prestosql.plugin.jdbc.ForBaseJdbc;
+import io.prestosql.plugin.jdbc.JdbcClient;
+import io.prestosql.plugin.jdbc.credential.CredentialProvider;
+import io.prestosql.spi.PrestoException;
+
+import java.lang.reflect.InvocationTargetException;
+import java.sql.Driver;
+
+import static io.airlift.configuration.ConfigBinder.configBinder;
+import static io.prestosql.plugin.jdbc.JdbcErrorCode.DRIVER_NOT_FOUND;
+
+public class GenericJdbcClientModule
+        implements Module
+{
+    @Override
+    public void configure(Binder binder)
+    {
+        binder.bind(Key.get(JdbcClient.class, ForBaseJdbc.class)).to(GenericJdbcClient.class)
+                 .in(Scopes.SINGLETON);
+        configBinder(binder).bindConfig(BaseJdbcConfig.class);
+    }
+
+    @Provides
+    @Singleton
+    @ForBaseJdbc
+    public ConnectionFactory getConnectionFactory(BaseJdbcConfig config, CredentialProvider credentialProvider)
+    {
+        try {
+            Class.forName(config.getDriverClass());
+        }
+        catch (ClassNotFoundException e) {
+            throw new PrestoException(DRIVER_NOT_FOUND, config.getDriverClass() + " not found");
+        }
+        try {
+            try {
+                return new DriverConnectionFactory((Driver) Class.forName(config.getDriverClass()).getConstructor().newInstance(), config, credentialProvider);
+            }
+            catch (InvocationTargetException e) {
+                e.printStackTrace();
+            }
+            catch (NoSuchMethodException e) {
+                e.printStackTrace();
+            }
+        }
+        catch (InstantiationException e) {
+            e.printStackTrace();
+        }
+        catch (IllegalAccessException e) {
+            e.printStackTrace();
+        }
+        catch (ClassNotFoundException e) {
+            e.printStackTrace();
+        }
+        return null;
```

I would throw an exception instead of null.